lib/chef/resource/support/client.erb

```diff
@@ -20,15 +20,15 @@
       @policy_group
       @policy_name
       @ssl_verify_mode).each do |prop| -%>
-<% next if instance_variable_get(prop).nil? || instance_variable_get(prop).empty? -%>
-<%=prop.delete_prefix("@") %> = <%= instance_variable_get(prop).inspect %>
+<% next if instance_variable_get(prop).nil? || instance_variable_get(prop).empty? || prop == '@log_location' -%>
+<%=prop.delete_prefix("@") %> <%= instance_variable_get(prop).inspect %>
 <% end -%>
 <%# log_location is special due to STDOUT/STDERR from String -> IO Object -%>
 <% unless @log_location.nil? %>
-  <% if @log_location.is_a? String && %w(STDOUT STDERR).include?(@log_location) -%>
-log_location = <%= @log_location %>
+  <% if (@log_location.is_a? String) && (%w(STDOUT STDERR).include?(@log_location)) -%>
```

Comment:
Is this line change necessary? Ruby should do the right thing with the order of operations without it.

Reply:
I agree it should, but without the parentheses, it throws the following:
```
  
      Chef::Mixin::Template::TemplateError
      ------------------------------------
      class or module required
```
I don't understand why this error is produced, but adding the parentheses stops erb getting upset.

Reply:
Maybe try @log_location.is_a?(String)

Reply:
Is it necessary to check that `@log_location` is a string, then immediately also check that it is equal to one of two other strings? We already know from the check above that it is not `nil`, so what other scenarios are we trying to guard against?

```suggestion
  <% if %w(STDOUT STDERR).include?(@log_location) -%>
```

Reply:
If this does need some more complicated logic it might be worth moving it into the resource where it could be potentially unit tested, etc, instead of embedded in the erb file.

Reply:
@phiggins to my knowledge the other scenarios are:
 `@log_location` can be a symbol as in `:syslog` or `:win_evt` or a path to a file, or the ruby constants STDOUT or STDERR

@tas50 `@log_location.is_a?(String)` works and I am happy the logic is simple enough to remain in the template